src/qt/transactionrecord.cpp

```diff
@@ -185,11 +185,18 @@ bool TransactionRecord::decomposeCreditTransaction(const CWallet* wallet, const
 {
     int64_t nTime = wtx.GetTxTime();
     const uint256& hash = wtx.GetHash();
+    unsigned int cachedTotalSize = 0;
     for (unsigned int nOut = 0; nOut < wtx.vout.size(); nOut++) {
         const CTxOut& txout = wtx.vout[nOut];
         isminetype mine = wallet->IsMine(txout);
         if (mine) {
-            TransactionRecord sub(hash, nTime, wtx.GetTotalSize());
+
+            // Cache total size to not be forced to recalculate it on every output
+            if (cachedTotalSize == 0) {
+                cachedTotalSize = wtx.GetTotalSize();
+            }
+
+            TransactionRecord sub(hash, nTime, cachedTotalSize);
```

Comment:
We can simply initialize the tx size when we declare it, outside of the loop, same as we do below for  `decomposeDebitTransaction`.
```cpp
unsigned int txSize = wtx.GetTotalSize();
```

Reply:
I did it in this way because it was calculating the size for nothing if there is no output from the wallet. 
But yeah.. thinking it more, `GetCredit` at the beginning of decomposeTransaction is being called with an `ISMINE_ALL` filter, which will prevent that from happening.